lib/subordinateio.c

```diff
@@ -307,39 +307,23 @@ static bool have_range(struct commonio_db *db,
 	return false;
 }
 
-static bool append_range(struct subordinate_range ***ranges, const struct subordinate_range *new, int n)
+static bool append_range(struct subordinate_range **ranges, const struct subordinate_range *in, int n)
 {
-	struct subordinate_range *tmp;
 	if (!*ranges) {
-		*ranges = malloc(sizeof(struct subordinate_range **));
+		*ranges = malloc(sizeof(struct subordinate_range));
 		if (!*ranges)
 			return false;
 	} else {
-		struct subordinate_range **new;
-		new = realloc(*ranges, (n + 1) * (sizeof(struct subordinate_range **)));
+		struct subordinate_range *new;
+		new = realloc(*ranges, (n + 1) * (sizeof(struct subordinate_range)));
 		if (!new)
 			return false;
 		*ranges = new;
 	}
-	(*ranges)[n] = NULL;
-	tmp = subordinate_dup(new);
-	if (!tmp)
-		return false;
-	(*ranges)[n] = tmp;
+	(*ranges)[n] = *in;
```

Comment:
This makes a copy of `subordinate_range::owner` pointer.
And it seems `commonio_next()` returns a pointer to internal structured (not copy).
So I guess dup of `subordinate_range::owner` is needed here?

Reply:
Btw, `subordinate_dup()` seems to be unused after this change.